nautobot/docs/user-guide/administration/upgrading/from-v1/ipam-upgrade-guide.md

```diff
@@ -0,0 +1,455 @@
+
+### Primary IPv4/IPv6 no longer unique
+
+On `Device` and `VirtualMachin`e objects, the `primary_ip4` and `primary_ip6` fields were changed from a one-to-one field--which is a foreign key with a uniqueness constraint--to a foreign key, dropping the uniquness constraint.
+
+This was necessary to support the case where the same `IPAddress` object may be assigned to one or more `Interface`/`VMInterface` objects, reducing the need to proliferate duplicate `IPAddress` objects merely for the purpose of facilitating `Interface`/`VMInterface` assignments.
+
+## Preparing your IPAM Data for Nautobot 2.0
+
+### Run the pre-migration helper before upgrading
+
+A new pre-migration management command was added that will allow you to check your existing data for compatibility with the data model changes introduced in Nautobot 2.0. You are highly encouraged to run this before upgrading!
+
+For more information please see the [documentation on Pre-migration validation](./upgrading-from-nautobot-v1.md##pre-migration-validation).
+
+### You cannot migrate Interfaces or VMInterfaces that have IPs with differing VRFs
+
+When assigning a `VRF` to an`IPaddress`, the `VRF` must be the same across each IP when multiple `IPAddress` objects are assigned to `Interface`/`VMInterface` objects.
+
+Make sure for `Interfaces`/`VMInterfaces` with multiple IPs that each `IPAddress`1 assigned to the same `Interface`/`VMInterface` share the same VRF, or are not assigned a VRF.
+
+### Parent Prefixes, child Prefixes, and child IPAddresses must share the same Namespace
+
+Parent `Prefixes`, child `Prefixes`, and `IPAddresses` must share the same `Namespace` and any case for duplicate `Prefix`/`IPAddress` must involve leveraging distinct `Namespace` objects.
+
+If you need to maintain duplicates for any reason, assert that each set of duplicate objects are assigned to a distinct `VRF` with `enforce_unique` set to `True`, as during the upgrade process these will each be moved to their own "VRF Namespace" (Please see the section on **VRF Namespaces** above).
+
+### IPAddress objects can no longer be orphaned
+
+The database migration will automatically create a parent `Prefix` for `IPAddresses` that do not have an eligible parent. For example an `IPAddress` with address of `1.2.3.4/32` will have a parent `Prefix` created of the same `network` and `prefix_length` e.g. `1.2.3.4/32`.
+
+If you do not wish for these single-host `Prefixes` to be created, create a parent `Prefix` of your desired size to contain any would-be orphaned `IPAddresses` before upgrading.
+
+## After you Upgrade
+
+This section includes various things to consider after you have successfully upgraded Nautobot.
+
+### Review any Cleanup Namespaces
+
+> This may also apply to any "VRF Namespace" objects that were created, depending on your requirements on maintaining duplicate Prefix/IPAddress objects.
```

Comment:
```suggestion
> This may also apply to any "VRF Namespace" objects that were created, depending on your requirements for maintaining duplicate Prefix/IPAddress objects.
```